src/compiler/crystal/compiler.cr

```diff
@@ -470,19 +470,31 @@ module Crystal
               pr, pw = IO.pipe
               spawn do
                 pr.each_line do |line|
-                  unit = JSON.parse(line)
-                  reused << unit["name"].as_s if unit["reused"].as_bool
+                  unit_info = JSON.parse(line)
+                  reused << unit_info["name"].as_s if unit_info["reused"].as_bool
+
+                  module_idx = unit_info["idx"].as_i
+                  slice.update(module_idx) do |unit|
+                    unit.compilation_time = unit_info["time"].as_f.seconds
+                    unit
+                  end
```

Comment:
`unit` is a reference, so I believe direct mutation works and there is no need to do it like a value

Reply:
Are you still going to address this?